src/server.h

```diff
@@ -1318,6 +1318,20 @@ typedef struct redisOpArray {
     int capacity;
 } redisOpArray;
 
+/* If a BGSAVE fails, we attempt to categorize its failure cause. */
+typedef enum {
+    SAVE_FAILURE_NONE=-1,
+    SAVE_FAILURE_CANCELED,
+    SAVE_FAILURE_OOM,
+    SAVE_FAILURE_NO_FDS,
+    SAVE_FAILURE_DISK,
+    SAVE_FAILURE_SOCKET,
+    SAVE_FAILURE_COB_OVERRUN_LARGE_ITEM,
```

Comment:
what's SAVE_FAILURE_COB_OVERRUN_LARGE_ITEM and SAVE_FAILURE_NO_FDS?
i don't see them being used.

Reply:
I agree.
SAVE_FAILURE_NO_FDS - should be returned in case we fail to create the pipe for the BGSAVE process. theoretically it could happen in case of limited number of fds
IDK why we thought we need the SAVE_FAILURE_COB_OVERRUN_LARGE_ITEM , but we can drop it.